storage/remote/queue_manager.go

```diff
@@ -1116,19 +1116,33 @@ func (q *queue) ReturnForReuse(batch []sampleOrExemplar) {
 // FlushAndShutdown stops the queue and flushes any samples. No appends can be
 // made after this is called.
 func (q *queue) FlushAndShutdown(done <-chan struct{}) {
+	for q.trySendingBatch(done) {
+		time.Sleep(time.Second)
+	}
```

Comment:
In practice I believe the sleep will only every be run 0 or 1 times. If we end up having to retry it is almost certainly because `(*queue).Batch()` is being called concurrently, and once we unlock `batchMtx` room on `batchQueue` will be made for the pending batch and the next time through will succeed. No harm in running more than once though.